web_advanced_search/static/src/js/web_advanced_search.js

```diff
@@ -0,0 +1,283 @@
+        }),
+
+        events: _.extend({}, FilterMenu.prototype.events, {
+            "click .o_add_advanced_search": "advanced_search_open",
+        }),
+
+        init: function () {
+            this._super.apply(this, arguments);
+            this.trigger_up("get_dataset");
+        },
+
+        /**
+         * Open advanced search dialog
+         *
+         * @returns {$.Deferred} The opening dialog itself.
+         */
+        advanced_search_open: function () {
+            var domain_selector_dialog = new DomainSelectorDialog(
+                this,
+                this.dataset.model,
+                "[]",
+                {
+                    debugMode: core.debug,
+                    readonly: false,
+                }
+            );
+            // Add 1st domain node by default
+            domain_selector_dialog.domainSelector._onAddFirstButtonClick();
+            return domain_selector_dialog.open();
+        },
+
+        /**
+         * Apply advanced search on dialog save
+         *
+         * @param {OdooEvent} event A `domain_selected` event from the dialog.
+         */
+        advanced_search_commit: function (event) {
+            _.invoke(this.propositions, "destroy");
+            var proposition = new AdvancedSearchProposition(
+                this,
+                event.data.domain
+            );
+            this.propositions = [proposition];
+            this.commit_search();
+        },
+    });
+
+    /**
+     * A search field for relational fields.
+     *
+     * It implements and extends the `FieldManagerMixing`, and acts as if it
+     * were a reduced dummy controller. Some actions "mock" the underlying
+     * model, since sometimes we use a char widget to fill related fields
+     * (which is not supported by that widget), and fields need an underlying
+     * model implementation, which can only hold fake data, given a search view
+     * has no data on it by definition.
+     */
+    var Relational = Char.extend(FieldManagerMixin, {
+        tagName: "div",
+        className: "x2x_container",
+        attributes: {},
+
+        init: function () {
+            this._super.apply(this, arguments);
+            // To make widgets work, we need to a model and an empty record
+            FieldManagerMixin.init.call(this);
+            this.trigger_up("get_dataset");
+            // Make equal and not equal appear 1st and 2nd
+            this.operators = _.sortBy(
+                this.operators,
+                function(op) {
+                    switch(op.value) {
+                        case "=":
+                            return -2;
+                        case "!=":
+                            return -1;
+                        default:
+                            return 0;
+                    }
+                });
+            // Create dummy record with only the field the user is searching
+            var params = {
```

Comment:
I'd move this to a separated function

Reply:
I have left this as it was. Thinking about it, it's only done once at init time, and calling it again could produce some problems, so it seems legit to not do a method about it IMHO.

I fixed your other comments though.